### ISSUE TITLE
Keycloak4s URL handling is a bit over-specified

### ISSUE THREAD

**Issue by Voltir** [fullfacing/keycloak4s#269]
My company is in the process of rolling out keycloak within a Kubernetes cluster. Due to some ingress controller routing issues, we had to override some of the routes keycloak uses - specifically, we replaced the "/auth" prefix with "/keycloak". We are also evaluating the use of keycloak4s to integrate our scala auth services with keycloak. However, the following to hard-coded path segments are somewhat problematic:

https://github.com/fullfacing/keycloak4s/blob/develop/keycloak4s-admin/src/main/scala/com/fullfacing/keycloak4s.admin/client/KeycloakClient.scala#L37

https://github.com/fullfacing/keycloak4s/blob/develop/keycloak4s-admin/src/main/scala/com/fullfacing/keycloak4s.admin/client/TokenManager.scala#L53

I hacked together a fork to verify that keycloak4s would work for us, aside from the above issue.

**Comment by StuartJ45** [comment]
Hi @Voltir ,
I have added a `basePath` field in the config class to allow for custom path prefix segments
Eg `https://keycloak.server.com:443/keycloak/admin/realms/master`:
```scala
  ConfigWithAuth(
    scheme   = "https",
    host     = "keycloak.server.com",
    port     = 443,
    realm    = "master",
    basePath = List("keycloak"),
    authn    = ???
  )
```
The feature has been published with version 2.3.0 :)

**Comment by Voltir** [comment]
ty!
